client/src/components/MapLegend.vue

```diff
@@ -21,32 +23,27 @@ import { defineComponent, PropType } from 'vue';
  */
 export default defineComponent({
   name: 'MapLegend',
+  setup() {
+    const state: State | undefined = inject(stateKey);
```

Comment:
Rather than allowing this to be `undefined`, is there a default state we could use?